ql/src/java/org/apache/hadoop/hive/ql/lockmgr/DbTxnManager.java

```diff
@@ -227,6 +228,12 @@ public long openTxn(Context ctx, String user, TxnType txnType) throws LockExcept
     return openTxn(ctx, user, txnType, 0);
   }
 
+  @Override
+  public long openTxn(Context ctx, String user, TxnType txnType, String dbUnderReplication) throws LockException {
```

Comment:
Can't we re-use replOpenTxn by enhancing that to take txnType?

Reply:
repl(Open/Rollback/Commit)Txn is specifically for replayed transaction. So, by convention it would be REPL_CREATED transactions. It would be confusion if we start introducing txnType in that.

Reply:
That's what I mean by enhancement. Like what if it takes txnType as well. Does that help?

Reply:
We need some tasks (like startHeartbeat, initialising lock manager, clearing some of variables) to be done in case of normal transactions which is not required in case of repl replayed transactions. So, it would be better if we skip this enhancement.

Reply:
We will just have two paths - one for repl created one for hive repl txn using same interface and they can differ in the payload(txnType/replPolicy/srcTxnIds.
In current form it would be confusing that which call is supposed to be used in repl and which is not.